src/options.js

```diff
@@ -0,0 +1,119 @@
+/*global chrome */
+function restore_options() {
+    loadDoNotReplaceList(function(doNotReplaceList) {
+        displayDoNotReplaceList(doNotReplaceList);
+    });
+}
+
+function loadDoNotReplaceList(callback) {
+    chrome.storage.sync.get(
+        {
+            doNotReplaceList: []
+        },
+        function(items) {
+            if (callback) {
+                callback(items.doNotReplaceList);
+            }
+        }
+    );
+}
+
+function saveDoNotReplaceList(doNotReplaceList, callback) {
+    const items = { doNotReplaceList: doNotReplaceList };
+    chrome.storage.sync.set(items, function(items) {
+        if (callback) {
+            callback();
+        }
+    });
+}
+
+function addItem() {
+    const newEntryField = document.getElementById("newEntry");
+
+    let url = newEntryField.value;
+
+    if (url.includes("://")) {
+        // Strip off the protocol using the first instance of '://'
+        url = url
+            .split("://")
+            .slice(1)
+            .join("://");
+    }
+
+    url = url.split("/")[0];
```

Comment:
The URL API requires a protocol, and since URLs in the list don't have protocols, we want to let the user submit a URL that doesn't have a protocol. I've added a comment line mentioning why.